### PR TITLE
spi: put a request buffer infront of SPI

### PR DESCRIPTION
**Pull request by terpstra**
This will prevent SPI from blocking other pbus requests.

### CODE REVIEW

**Comment by aswaterman** [comment]
I think this is the right thing to do, but for completeness, I'll mention that it won't address a recent failure due to a TL monitor timeout, because two requests to SPI were convoyed.

**Comment by terpstra** [comment]
Sure. This only helps cross-traffic make progress when SPI is active. It won't help if you actually need to read from the SPI.

**Comment by terpstra** [comment]
Hmm. I just realized that TLBuffer should probably look if all the slaves are read-only, and if so, push a constant zero for the data in the Q. Not going to fix this for aloe v0, tho.

**Comment by terpstra** [comment]
Actually, there is a WidthWidget down to 1 byte before the TLBuffer. 'Get' is single-beat no matter how large the request, so the data bits wasted are pretty insignificant.

File: src/main/scala/devices/spi/SPIPeriphery.scala [hunk 1]
```diff
@@ -41,7 +41,10 @@ trait HasPeripherySPIFlash extends HasPeripheryBus with HasInterruptBus {
   val qspis = spiFlashParams map { params =>
     val qspi = LazyModule(new TLSPIFlash(pbus.beatBytes, params))
     qspi.rnode := pbus.toVariableWidthSlaves
-    qspi.fnode := TLFragmenter(1, pbus.blockBytes)(pbus.toFixedWidthSlaves)
+    qspi.fnode :=
+      TLFragmenter(1, pbus.blockBytes)(
+      TLBuffer(BufferParams(8), BufferParams.none)(
```
Comment (mwachs5) [inline src/main/scala/devices/spi/SPIPeriphery.scala]: suggest we make this a config parameter in the spiFlashParams

Comment (terpstra) [inline src/main/scala/devices/spi/SPIPeriphery.scala]: Agreed. But, did it fix the problem?

**Pull request merged**